### PR TITLE
Add Globalnet v2 gateway monitor

### PR DESCRIPTION
**Pull request by tpantelis**
Based on the previous one and starts the `GlobalEgressIP` and `ClusterGlobalEgressIP` controllers.

### CODE REVIEW

File: pkg/globalnet/controllers/gateway_monitor.go [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+	gatewayMonitor := &gatewayMonitor{
+		baseController: newBaseController(),
+		spec:           spec,
+		isGatewayNode:  false,
+		remoteSubnets:  stringset.NewSynchronized(),
+	}
+
+	var err error
+
+	gatewayMonitor.ipt, err = iptables.New()
```
Comment (mangelajo) [inline pkg/globalnet/controllers/gateway_monitor.go:56]: Can we use the event handler architecture in here?.

In the case of OVN we have the chance of eventually creating an OVN-specific version of the handler which would not be iptables based. Also would smoothen the transition to other architectures like nftables eventually.

Comment (mangelajo) [inline pkg/globalnet/controllers/gateway_monitor.go:56]: Can we use the event handler architecture in here?.

In the case of OVN we have the chance of eventually creating an OVN-specific version of the handler which would not be iptables based. Also would smoothen the transition to other architectures like nftables eventually.

**Review by mangelajo — CHANGES_REQUESTED** [review]
@sridhargaddam @tpantelis some comments about this. Is it possible to construct the gateway monitor like an event handler? 

https://github.com/submariner-io/submariner/blob/devel/pkg/routeagent_driver/main.go#L84

File: pkg/globalnet/controllers/gateway_monitor.go (hunk 1 above)
Comment (sridhargaddam) [reply pkg/globalnet/controllers/gateway_monitor.go:56]: Right, we are also thinking of using the Event handler framework and based on TransitionToGateway and TransitionToNonGateway, we can start/stop the other Globalnet V2 controllers

Comment (tpantelis) [reply pkg/globalnet/controllers/gateway_monitor.go:56]: yes that is the [plan](https://github.com/submariner-io/submariner/issues/1331). The new controllers are higher priority - just want to get those working first.

Comment (mangelajo) [reply pkg/globalnet/controllers/gateway_monitor.go:56]: ack :+1:

**Review by mangelajo — APPROVED** [review]
ok I'm fine with moving along, as long as we move into event handler before releasing

**Review by sridhargaddam — APPROVED** [review]

**Pull request merged**